### PR TITLE
Add release_date sort

### PR DESCRIPTION
**Pull request by AndreasPB**
# Description

* Adds the ability to sort by release date
* Adds tests for release_date and popularity
* Removes popularity as the default sorting - We are back to the custom ranking rules

Part of #248 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by Primdahl26 — CHANGES_REQUESTED** [review]
Dont we need to add this to our `Meilisearch` endpoint? How would we query it through the API otherwise?

**Comment by AndreasPB** [comment]
> Dont we need to add this to our `Meilisearch` endpoint? How would we query it through the API otherwise?

Derp, that went a little too fast :D 

I went all out with Swagger documentation and enums. I also pulled popularity out as an option, so we are back to the "real" MeiliSearch way with the ranking rules we have set up.

Side note for the frontend: You wont get much out of having to sorts on at the same time, so that should probably be prevented

File: frontend/src/components/filters.svelte [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+  $: {
+    if ($sorting.byPopularity) {
+      descLabelText = "Highest"
+      ascLabelText = "Lowest"
+    } else {
+      descLabelText = "Newest"
+      ascLabelText = "Oldest"
+    }
+  }
+
```
Comment (AndreasPB) [inline frontend/src/components/filters.svelte:27]: Can we make it not use the popularity sorting when the user has input(unless it was actively selected)

File: frontend/src/routes/index.svelte [hunk 2]
```diff
@@ -85,6 +86,14 @@
...
       query += `&only_providers=true`
     }
+
+    if (!input) {
+      if ($sorting.byPopularity) {
+        query += `&popularity=${$sorting.asc ? "asc" : "desc"}`
+      } else if ($sorting.byReleaseDate) {
+        query += `&release_date=${$sorting.asc ? "asc" : "desc"}`
+      }
+    }
```
Comment (AndreasPB) [inline frontend/src/routes/index.svelte:96]: Can we have a third option(byDefault(?)) that does this with the !input and have the old behavior on byPopularity and byReleaseDate?

**Review by nullxDEADBEEF — APPROVED** [review]
SGUFM :shipit:

**Review by Primdahl26 — APPROVED** [review]

**Pull request merged**